pkg/backup/backup_manager.go

```diff
@@ -83,13 +86,69 @@ func (bm *BackupManager) SaveSnap(ctx context.Context, s3Path string, isPeriodic
 }
 
 // EnsureMaxBackup to ensure the number of snapshot is under maxcount
+// We need to naturally sort it, help functions below:
+type Compare func(str1, str2 string) bool
+
+func (cmp Compare) Sort(strs []string) {
+	strSort := &strSorter{
+		strs: strs,
+		cmp:  cmp,
+	}
+	sort.Sort(strSort)
+}
+
+type strSorter struct {
+	strs []string
+	cmp  func(str1, str2 string) bool
+}
+
+func extractNumberFromString(str string, size int) (num int) {
```

Comment:
Might be worth mentioning where you got his function from?  Appears to be a direct copy from https://www.socketloop.com/tutorials/golang-natural-string-sorting-example

Reply:
yeah, fair enough. What is the etiquette here? A comment and the link.